### PR TITLE
feat: add write off page

### PR DESCRIPTION
**Pull request by BLasan**
## Description
Write Off page component added.

## Related issues and discussion
#847 

## Screenshots, if any
Screenshot from 2020-06-11 18-58-28

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

**Comment by BLasan** [comment]
@muskankhedia Please review

File: src/app/loans/common-resolvers/loan-transaction-writeoff.resolver.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/** Angular Imports */
```
Comment (muskankhedia) [inline src/app/loans/common-resolvers/loan-transaction-writeoff.resolver.ts:1]: @BLasan Please use the common resolver `loan-action-button.resolver.ts` for all the actions with the 'if condition'.

File: src/app/loans/loans-view/loan-account-actions/write-off-page/write-off-page.component.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    const transactionDate = this.writeOffForm.value.transactionDate;
+    const dateFormat = 'dd MMMM yyyy';
+    this.writeOffForm.patchValue({
+      transactionDate: this.datePipe.transform(transactionDate, dateFormat)
+    });
+    const writeOffForm = this.writeOffForm.value;
+    delete writeOffForm.amount;
+    writeOffForm.locale = 'en';
+    writeOffForm.dateFormat = dateFormat;
+    this.loanService.writeOffLoanTransaction(this.transactionData.loanData.id, writeOffForm).subscribe((response: any) => {
```
Comment (muskankhedia) [inline src/app/loans/loans-view/loan-account-actions/write-off-page/write-off-page.component.ts:71]: The common function `loanActionButtons()` can be used.

File: src/app/loans/loans-view/loan-account-actions/write-off-page/write-off-page.component.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          <mat-datepicker-toggle matSuffix [for]="writeOffDatePicker"></mat-datepicker-toggle>
+          <mat-datepicker #writeOffDatePicker></mat-datepicker>
+          <mat-error *ngIf="writeOffForm.controls.transactionDate.hasError('required')">
+            Write Off Date <strong>is required</strong>
+          </mat-error>
+        </mat-form-field>
+  
+        <mat-form-field>
+          <mat-label>Amount</mat-label>
+          <input matInput required formControlName="amount">
```
Comment (muskankhedia) [inline src/app/loans/loans-view/loan-account-actions/write-off-page/write-off-page.component.html:21]: `Amount` isn't a **required** field. Please change it.

**Comment by BLasan** [comment]
@muskankhedia Done. Please review

File: src/app/loans/loans-view/loan-account-actions/loan-account-actions.component.html [hunk 4]
```diff
@@ -7,5 +7,6 @@
     <mifosx-prepay-loan [dataObject]="actionButtonData" *ngIf="actions.prepay_loan"></mifosx-prepay-loan>
     <mifosx-make-repayment [dataObject]="actionButtonData" *ngIf="actions.make_repayment"></mifosx-make-repayment>
     <mifosx-waive-interest [dataObject]="actionButtonData" *ngIf="actions.waive_interest"></mifosx-waive-interest>
+    <mifosx-write-off-page [dataObject]="actionButtonData" *ngIf="actions"></mifosx-write-off-page>
```
Comment (muskankhedia) [inline src/app/loans/loans-view/loan-account-actions/loan-account-actions.component.html:10]:
```suggestion
    <mifosx-write-off-page [dataObject]="actionButtonData" *ngIf="actions.write_off"></mifosx-write-off-page>
```

File: src/app/loans/loans-view/loan-account-actions/write-off-page/write-off-page.component.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  styleUrls: ['./write-off-page.component.scss']
+})
+export class WriteOffPageComponent implements OnInit {
+
+  @Input() dataObject: any;
+
+  /** Write Off form. */
+  writeOffForm: FormGroup;
+  /** Transaction data. */
+  transactionData: any = new Object();
```
Comment (muskankhedia) [inline src/app/loans/loans-view/loan-account-actions/write-off-page/write-off-page.component.ts:24]: Please remove this, if not required.

**Comment by BLasan** [comment]
@muskankhedia  Please review

**Pull request merged**